lib/src/RedisRemoteSaiInterface.cpp

```diff
@@ -373,9 +374,46 @@ sai_status_t RedisRemoteSaiInterface::setRedisExtensionAttribute(
 
             return SAI_STATUS_SUCCESS;
 
+        case SAI_REDIS_SWITCH_ATTR_ZMQ_SYNC_MODE:
+
+            m_zmqSyncMode = attr->value.booldata;
+
+            if (m_zmqSyncMode)
+            {
+                m_contextConfig->m_zmqEnable = m_zmqSyncMode;
+
+                // main communication channel was created at initialize method
+                // so this command will replace it with zmq channel
+
+                m_communicationChannel = std::make_shared<ZeroMQChannel>(
+                        m_contextConfig->m_zmqEndpoint,
+                        m_contextConfig->m_zmqNtfEndpoint,
+                        std::bind(&RedisRemoteSaiInterface::handleNotification, this, _1, _2, _3));
+
+                SWSS_LOG_NOTICE("zmq enabled, forcing sync mode");
+
+                m_syncMode = true;
```

Comment:
>m_syncMode [](start = 16, length = 10)

Does it make more sense to return SAI_STATUS_NOT_SUPPORTED if m_zmqSyncMode==True and m_syncMode==False?
This way the caller will know there is a logic error.

Reply:
actually zmq syncd mode forces sync mode internally, so i don't think there is error here